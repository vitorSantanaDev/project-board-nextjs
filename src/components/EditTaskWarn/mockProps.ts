export const handleClickCancelEdit = jest.fn()
