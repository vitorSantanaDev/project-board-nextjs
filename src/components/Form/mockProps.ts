export const formProps = {
  handleAddTaskSubmit: jest.fn(),
  handleChangeInput: jest.fn(),
  input: ''
}
