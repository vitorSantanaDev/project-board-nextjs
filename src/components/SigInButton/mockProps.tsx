export const mockProps = {
  session: true,
  onClick: jest.fn()
}
