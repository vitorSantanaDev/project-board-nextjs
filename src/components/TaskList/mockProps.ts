export const tasksProps = [
  {
    id: '3896433',
    userId: '8458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '39033',
    userId: '8456458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '38933',
    userId: '865658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '3366553',
    userId: '85654658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '337763',
    userId: '88',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  }
]
