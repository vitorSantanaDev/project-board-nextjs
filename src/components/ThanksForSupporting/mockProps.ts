export const mockProps = {
  lastDonateUser: `Sua última doação foi a ${new Date().toLocaleDateString()}`
}
