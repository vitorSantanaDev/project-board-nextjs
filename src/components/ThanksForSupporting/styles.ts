import styled from 'styled-components'

export const Wrapper = styled.div`
  max-width: 1120px;
  border-radius: 6px;

  padding: 2rem;
  margin: 2rem auto;

  background-color: #20212c;
`

export const Message = styled.h3`
  color: #ffb800;
  font-size: 2.8rem;
`

export const TimeOfLastDonation = styled.div`
  margin-top: 3rem;

  display: flex;
  align-items: center;

  > time {
    color: #fff;
    font-size: 1.8rem;

    padding: 1rem;
  }
`
