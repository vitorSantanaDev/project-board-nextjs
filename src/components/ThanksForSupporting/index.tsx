import { FiClock } from 'react-icons/fi'
import { formatDistance } from 'date-fns'
import { ptBR } from 'date-fns/locale'

import * as S from './styles'

interface ThanksForSupportingProps {
  lastDonateUser: string | Date
}

const ThanksForSupporting: React.FC<ThanksForSupportingProps> = ({
  lastDonateUser
}) => {
  return (
    <S.Wrapper>
      <S.Message>Obrigado por apoiar esse projeto.</S.Message>
      <S.TimeOfLastDonation>
        <FiClock size={28} color="#FFF" />
        <time>
          Última doação foi a{' '}
          {formatDistance(new Date(lastDonateUser), new Date(), {
            locale: ptBR
          })}
        </time>
      </S.TimeOfLastDonation>
    </S.Wrapper>
  )
}

export default ThanksForSupporting
