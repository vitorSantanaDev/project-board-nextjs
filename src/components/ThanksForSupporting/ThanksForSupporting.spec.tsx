import { render, screen } from '@testing-library/react'
import ThanksForSupporting from '.'
import { mockProps } from './mockProps'

describe('<ThanksForSupporting />', () => {
  it('should render the heading', () => {
    render(<ThanksForSupporting {...mockProps} />)
    const heading = screen.getByRole('heading', {
      name: /obrigado por apoiar esse projeto\./i
    })
    const dateOfDonators = screen.getByText(/última doação foi a/i)

    expect(heading).toBeInTheDocument()
    expect(dateOfDonators).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<ThanksForSupporting {...mockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
