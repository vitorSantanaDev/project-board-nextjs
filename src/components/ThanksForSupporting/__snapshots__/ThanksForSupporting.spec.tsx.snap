// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThanksForSupporting /> should match snapshot 1`] = `
.c0 {
  max-width: 1120px;
  border-radius: 6px;
  padding: 2rem;
  margin: 2rem auto;
  background-color: #20212c;
}

.c1 {
  color: #ffb800;
  font-size: 2.8rem;
}

.c2 {
  margin-top: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > time {
  color: #fff;
  font-size: 1.8rem;
  padding: 1rem;
}

@media only screen and (max-width:1024px) {
  .c0 {
    margin: 2rem 2rem;
  }
}

<div
  class="c0"
>
  <h3
    class="c1"
  >
    Obrigado por apoiar esse projeto.
  </h3>
  <div
    class="c2"
  >
    <svg
      color="#FFF"
      fill="none"
      height="28"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(255, 255, 255);"
      viewBox="0 0 24 24"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <polyline
        points="12 6 12 12 16 14"
      />
    </svg>
    <time>
      Última doação foi a
       
      35 minutos
    </time>
  </div>
</div>
`;
